constants/src/lib.rs

```diff
@@ -82,6 +82,10 @@ pub const UNABLE_TO_CONNECT: OdbcState<'static> = OdbcState {
     odbc_3_state: "08001",
 };
 pub const INVALID_DESCRIPTOR_INDEX: OdbcState<'static> = OdbcState {
+    odbc_2_state: "S1002",
+    odbc_3_state: "07009",
+};
+pub const INVALID_COLUMN_NUMBER: OdbcState<'static> = OdbcState {
```

Comment:
I split off a separate error, and made the existing error map to S1002 because it's used far more. This is what I found, let me know if this makes sense. INVALID_DESCRIPTOR_INDEX was used by two errors:

- `ColIndexOutOfBounds`: used in `get_value` functions for `MongoCollections`, `MongoDatabases`, `MongoFields`, `MongoTableTypes`, and `MongoTypesInfo`, which then only get called by `SQLGetData` (and thus should return `S1002`). Also used by `MongoStatement.get_col_metadata` which is called in `SQLColAttribute` and `SQLDescribeCol`, but I don't think the error is ever directly used
- `InvalidDescriptorIndex`: error that ends up getting thrown when `MongoStatement.get_col_metadata` does not return ok; updated to the new error for `SQLDescribeCol` so that it uses the 07009 sql state; use the existing error for `SQLColAttribute` so that the sql state is `S1002`